Documentation/policy/language.rst

```diff
@@ -837,6 +914,75 @@ Allow producing to topic empire-announce using apiKeys
 
         .. literalinclude:: ../../examples/policies/l7/kafka/kafka.json
 
+DNS Policy and IP Discovery
+---------------------------
+
+Policy may be applied to DNS traffic, allowing or disallowing specific DNS
+query names or patterns of names (other DNS fields, such as query type, are not
+considered). This policy is effected via a DNS proxy, and is also used to
+collect IPs to be used in L3 `DNS based`_ ``toFQDNs`` rules.
+
+.. note::  While Layer 7 DNS policy can be applied without any other Layer 3
+           rules, the presence of a Layer 7 rule (with it's Layer 3 and 4
+           components) will block other traffic.
+
+DNS policy may be applied via:
+
+``matchName``
+  Allows queries for domains that match ``matchName`` exactly. Multiple
+  distinct names may be included in separate ``matchName`` entries and queries
+  for domains that match any ``matchName`` will be allowed.
+
+``matchPattern``
+  Allows queries for domains that match the pattern in ``matchPattern``,
+  accounting for wildcards. Patterns are composed of literal characters that
+  that are allowed in domain names: a-z, 0-9, ``.`` and ``-``.
+
+  ``*`` is allowed as a wildcard with a number of convenience behaviors:
+
+  * ``*`` within a domain allows 0 or more valid DNS characters, except for the
+    ``.`` separator. ``*.cilium.io`` will match ``sub.cilium.io`` but not
+    ``cilium.io``. ``part*ial.com`` will match ``partial.com`` and
+    ``part-extra-ial.com``.
+  * ``*`` at the start of a pattern, with no trailing ``.``, matches a domain
+    and all subdomains. For example, ``*cilium.io`` matches ``sub.cilium.io``
+    and ``cilium.io`` but not ``another.sub.cilium.io``.
+  * ``*`` alone matches all names, and inserts all IPs in DNS responses into
+    the ``cilium-agent`` DNS cache.
+
+In this example, L7 DNS policy allows queries for ``cilium.io`` and any
+subdomains of ``cilium.io`` and ``api.cilium.io``. No other DNS queries will be
+allowed.
+
+The separate L3 ``toFQDNs`` egress rule allows connections to any IPs returned
+in DNS queries for ``cilium.io``, ``sub.cilium.io``, ``service1.api.cilium.io``
+and any matches of ``special*service.api.cilium.io``, such as
+``special-region1-service.api.cilium.io`` but not
+``region1-service.api.cilium.io``. DNS queries to ``anothersub.cilium.io`` are
+allowed but connections to the returned IPs are not, as there is no L3 toFQDNs
```

Comment:
should this be \`\`toFQDNs\`\` (surrounded by backticks)?

Reply:
yup! thanks for spotting that (you actually read the words! ;) )